modules/tinymce/src/core/main/ts/api/html/Styles.ts

```diff
@@ -263,6 +265,13 @@ const Styles = (settings: StylesSettings = {}, schema?: Schema): Styles => {
               value = value.toLowerCase();
             }
 
+            // Convert RGB colors to HEX
+            if (settings.force_hex_color) {
+              RgbaColour.fromString(value).each((rgba) => {
+                value = Fun.pipe(rgba, RgbaColour.toString, Transformations.rgbaToHexString);
```

Comment:
I don't think you need to use `Fun.pipe` here I think that was mostly created for the `Maybe` that we are not using anyware to make it easier to work with partially applied functions. So if you have functions that looks like this: `const foo = (something) => (a) => ..` then it would look nice if you pipe them like `Fun.pipe(a, foo(something))` instead of `foo(something)(a)` and if you have many of them its even more ugly.
But in this case I don't think this reads all that bad:
```suggestion
                value = Transformations.rgbaToHexString(RgbaColour.toString(rgba));
```

Reply:
Another issue is this looks like it's throwing away the alpha channel. I wonder if we should keep those so we only force hex colors if we really can force hex colors. Or should there be a mode to handle that as as well like a less forceful mode? If the alpha channel is fully opaque then we can do this operation safely without information loss.

Reply:
There are specific tests for this module in the StylesTest.ts test for this test should be there as well since the ColorPickerTest seems more like a integration test more higher up at UI level. Also add a case for the alpha channels so that it shows clearly that it does throw away that channel and for what cases.

Reply:
> I don't think you need to use `Fun.pipe` here I think that was mostly created for the `Maybe` that we are not using anyware to make it easier to work with partially applied functions. So if you have functions that looks like this: `const foo = (something) => (a) => ..` then it would look nice if you pipe them like `Fun.pipe(a, foo(something))` instead of `foo(something)(a)` and if you have many of them its even more ugly. But in this case I don't think this reads all that bad:

Makes sense, I removed the `Fun.pipe`

Reply:
> Another issue is this looks like it's throwing away the alpha channel. I wonder if we should keep those so we only force hex colors if we really can force hex colors. Or should there be a mode to handle that as as well like a less forceful mode? If the alpha channel is fully opaque then we can do this operation safely without information loss.

So would you maybe prefer a union type for the option, like: `force_hex_color: 'always' | 'rgb_only' | 'off';`?
Or split it into two options like `force_hex_color` & `coerce_hex_color`, or something else?

Reply:
Yes I think we should bring this back in a better version that what we had before. When the original option was created that forced hex colors rgba didn't exist, only rgb, as css concept so then there was no need to consider that.

Also not having a relax option could then cause new bug reports of users pasting something then the colors change because we strip the alpha channel. Not that I remember ever seeing that when we had that option so might be rare.

I think it's better to have one single option since they would otherwise be hard to explain what happens if you set one or the other or both etc.

Reply:
Ok, I've made those changes. There's some tests that attempt to cover using RGBA too

Reply:
So much for my idea of removing this option in TinyMCE 7 😂 

We need to update this code:
https://github.com/tinymce/tinymce/blob/14da33f3d1e666bea45a4bd392073a355803e4ca/modules/acid/src/main/ts/ephox/acid/api/colour/HexColour.ts#L39

To support hex with alpha, which all supported browsers have had for many years:
https://developer.mozilla.org/en-US/docs/Web/CSS/hex-color

So I guess in v7 the ticket should be "support alpha in the rgb to hex colour code, _then_ delete the force hex colour option"

Reply:
It doesn't work on IE and we don't support IE for tinymce anymore, however do we or should we support IE as an output target. I can't of the top of my head think of any html that we produce tight now that wouldn't work on IE but I guess that will change overtime since we don't test on IE. I guess we could go with hex rgba and see if we get complaints and then make some backwards compatible option since this is an opt in we could document this as a good to know kind of thing like don't enable this if you have an target audience that uses IE.

Reply:
Ah, I see what you mean. I think for TinyMCE 7 we should be safe saying we don't create IE-compatible HTML anymore. But that's a discussion for another day 😂